cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -630,27 +629,22 @@ public boolean meetCompletenessRequirements(List<String> goalNames) {
    * Get a goals by priority based on the goal list.
    *
    * @param goals A list of goals.
-   * @return A map of goal priority to goal.
+   * @return A list of goals sorted by highest to lowest priority.
    */
-  private Map<Integer, Goal> goalsByPriority(List<String> goals) {
+  private List<Goal> goalsByPriority(List<String> goals) {
     if (goals == null || goals.isEmpty()) {
       return AnalyzerUtils.getGoalMapByPriority(_config);
     }
     Map<String, Goal> allGoals = AnalyzerUtils.getCaseInsensitiveGoalsByName(_config);
```

Comment:
Good point, we don't need a `TreeMap` there.

Reply:
Ah, this is because we are making the comparison on goal names as case insensitive via `TreeMap<>(String.CASE_INSENSITIVE_ORDER);`. In other words, we would like to ensure that the `GoalName == goalNaMe` is `true`.